proxy/src/main/java/com/velocitypowered/proxy/connection/ConnectionType.java

```diff
@@ -0,0 +1,69 @@
+package com.velocitypowered.proxy.connection;
+
+import com.velocitypowered.api.util.GameProfile;
+import com.velocitypowered.proxy.config.PlayerInfoForwarding;
+import com.velocitypowered.proxy.connection.backend.BackendConnectionPhase;
+import com.velocitypowered.proxy.connection.client.ClientConnectionPhase;
+import com.velocitypowered.proxy.connection.forge.legacy.LegacyForgeConnectionType;
+import com.velocitypowered.proxy.connection.util.ConnectionTypeImpl;
+
+/**
+ * The types of connection that may be selected.
+ */
+public interface ConnectionType {
+
+  /**
+   * Indicates that the connection has yet to reach the
+   * point where we have a definitive answer as to what
+   * type of connection we have.
+   */
+  ConnectionType UNDETERMINED =
```

Comment:
I would strongly recommend these to be factored out into a `ConnectionTypes` class. Keeping these in the interface clutters things up.